Add build and ci support for cdn

rayepps opened the pull request:
## Description
To prepare for hosting radash on a CDN:
- updated the rollup config to build a `cdn` directory with `radash.js` (csm), `radash.min.js` (csm), and `radash.esm.js` (es)
- updated the build script to write a `.version` file to the `cdn` directory so we can assert that the build was run in CI/CD
- updated the pr validation workflow to check that the `cdn` directory has an updated build that matches the current code version
- updated the pr template to add a checkbox that reminds contributors to run the build (to produce the updated `cdn` directory)

## Checklist
- [ ] Changes are covered by tests if behavior has been changed or added
- [ ] Tests have 100% coverage
- [ ] The version in `package.json` has been bumped according to the changes made and standard semantic versioning rules

## Resolves
Resolves #140

rayepps commented:
Hey @matteosacchetto can you check on the `rollup.config.js` changes I made here? You're kind of the expert on the rollup script.

rayepps commented:
So this doesn't work :) need to figure out how to get rollup to warp the cdn build files in an IIFE for use in the browser.

rayepps commented:
Now it works :)

matteosacchetto commented:
Thank you for reaching out!
So, I have a bunch of suggestions related to the files creation for CDN!

I would suggest keeping the CDN build separate from the CJS and ESM builds and I would suggest to use the minify function of esbuild to minify the code (so that we do not have an old dependency)

matteosacchetto commented:
This is the config I would use for this case:
```js
// rollup.config.mjs
import typescript from '@rollup/plugin-typescript'
import dts from 'rollup-plugin-dts'
import esbuild, { minify } from 'rollup-plugin-esbuild'
import externals from 'rollup-plugin-node-externals'

const usePreferConst = true // Use "const" instead of "var"
const usePreserveModules = true // `true` -> keep modules structure, `false` -> combine everything into a single file
const useStrict = true // Use "strict"
const useThrowOnError = true // On error throw and exception
const useSourceMap = true // Generate source map files
const useEsbuild = true // `true` -> use esbuild, `false` use tsc

export default [
  {
    // .d.ts build
    input: 'src/index.ts',
    output: {
      file: 'dist/index.d.ts',
      format: 'es'
    },
    plugins: [externals(), dts()]
  },
  {
    // CJS build
    input: 'src/index.ts',
    output: {
      dir: 'dist/cjs',
      format: 'cjs',
      generatedCode: {
        constBindings: usePreferConst
      },
      preserveModules: usePreserveModules,
      strict: useStrict,
      entryFileNames: '[name].cjs',
      sourcemap: useSourceMap
    },
    plugins: [
      externals(),
      useEsbuild
        ? esbuild()
        : typescript({
          noEmitOnError: useThrowOnError,
          outDir: 'dist/cjs',
          removeComments: true
        })
    ]
  },
  {
    // ESM builds
    input: 'src/index.ts',
    output: {
      dir: 'dist/esm',
      format: 'es',
      generatedCode: {
        constBindings: usePreferConst
      },
      preserveModules: usePreserveModules,
      strict: useStrict,
      entryFileNames: '[name].mjs',
      sourcemap: useSourceMap
    },
    plugins: [
      externals(),
      useEsbuild
        ? esbuild()
        : typescript({
          noEmitOnError: useThrowOnError,
          outDir: 'dist/esm',
          removeComments: true
        })
    ]
  },
  {
    // CDN build
    input: 'src/index.ts',
    output: [{
      format: 'iife',
      generatedCode: {
        constBindings: usePreferConst
      },
      preserveModules: false,
      strict: useStrict,
      file: 'cdn/radash.js',
      name: 'radash',
      sourcemap: false
    }, {
      format: 'iife',
      generatedCode: {
        constBindings: usePreferConst
      },
      preserveModules: false,
      strict: useStrict,
      file: 'cdn/radash.min.js',
      name: 'radash',
      sourcemap: false,
      plugins: [minify()]
    }, {
      format: 'es',
      generatedCode: {
        constBindings: usePreferConst
      },
      preserveModules: false,
      strict: useStrict,
      file: 'cdn/radash.esm.js',
      sourcemap: false
    }],
    plugins: [
      externals(),
      useEsbuild
        ? esbuild()
        : typescript({
          noEmitOnError: useThrowOnError,
          outDir: 'cdn',
          removeComments: true
        })
    ]
  }
]
```

matteosacchetto commented:
This way would do not have to add terser as a dependency.
Let me know what you think about it 😃

matteosacchetto commented on package.json:
```diff
@@ -23,7 +23,8 @@
     "check": "yarn lint && yarn test && yarn build",
-    "build": "yarn tsc --noEmit && yarn rollup -c",
+    "build": "yarn tsc --noEmit && yarn rollup -c && yarn build:cdn:version",
+    "build:cdn:version": "jq -r .version package.json > cdn/.version",
```
This could be done also with `echo ${npm_package_version} > cdn/.version`, if you do want to avoid `jq` as a build dependency.

matteosacchetto reviewed (changes requested):
Check the rollup config I posted in the comments. It think it may be easier to manage and avoid introducing "old" dependencies

rayepps commented:
Thank you @matteosacchetto for the feedback! I love what you did with the rollup config and the `${npm_package_version}` (somehow I've never seen that). I implemented both and dropped the terser plugin. It's much better with those changes so thank you 🍻

matteosacchetto commented:
I was wondering, why do you keep track of the CDN files in the repo?

matteosacchetto commented:
Ok, got it! It seems like the easiest way to manage it! 😃
